src/ai/susi/mind/SusiSkill.java

```diff
@@ -625,7 +625,10 @@ public static int getSkillUsage(String model, String group, String language, Str
                         JSONArray skillName = languageName.getJSONArray(skillname);
 
                         // Fetch skill usage data for sorting purpose.
-                        int startIndex = skillName.length() >= duration ? skillName.length()-duration : 0;
+                        int startIndex = 0;
+                        if(duration >= 0) {
+                            startIndex = skillName.length() >= duration ? skillName.length()-duration : 0;
+                        }
```

Comment:
This will make the API complex. For every call in which duration parameter is not given the loop will run over the whole JSONArray unnecessarily.

Reply:
This is done, so that we can get an overall usage stats. Currently, we do not know the duration that has to be entered to get the overall stats. For duration based filtering, we can always add the duration parameter. Please correct me, if I am wrong.

Reply:
Suppose we don't want usage based sorting, but according to these changes it will still access the whole usage data. For fetching the whole usage data we can send a big integer in the duration parameter.

Reply:
@AnupKumarPanwar If we don't want usage based sorting, what should be the duration value in that case?
Also, if we send a big number in duration param, won't that be same as running the entire loop. Also, it wouldn't give us the result of the entire usage data, if it exceeds that big number. 

I agree to the time complexity issue, raised by you. But, for that, we need to make modifications in the way, we store usage data. Because, getting the highest used skills, considering the lifetime usage is an important indicator for us, wrt to the stats and analytics part.

Reply:
@akshatnitd If we do not want a usage based sorting then we will not send the duration parameter. According to the existing code it will make the duration=0 and startIndex = skillName.length(). Which means that the loop to count usage data will run 0 times.
But according to your PR, by default duration = -1 and startIndex=0, which means the loop will run over whole data.

Reply:
@AnupKumarPanwar I think you are talking about the case of `/getSkillList.json` API call. Currently, it shows a usage count of `0` for all skills, as duration isn't passed. Shouldn't we instead remove it from response, rather than showing `0` ?

Reply:
@AnupKumarPanwar I think you are talking about the case of `/getSkillList.json` API call. Currently, it shows a usage count of `0` for all skills, as duration isn't passed. Shouldn't we instead remove it from response, rather than showing `0` ?

Reply:
Yes we can remove it from the response when duration parameter isn't passed. When the client is not interested in the usage count, it will not pass duration parameter in the API call. So considering it as 0 will reduce the unnecessary complexity. We can remove it too when the duration is zero.